gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2

```diff
@@ -1210,6 +1212,61 @@ def test_{{ method_name }}_rest(transport: str = 'rest', request_type={{ method.
     {% endif %}
 
 
+    {% if method.input.required_fields %}
```

Comment:
It might be reasonable to have a test or two just for the hidden required fields update method.

Reply:
I'm not sure what you mean.  Which method is hidden?

Reply:
Oh, I see, once we refactor as you describe above.
This is quite tricky to test, which is a lot of motivation for these changes.  The problem is that typically, though not always, a required field is going to have an expected template in the http rule, so the default value will cause the transcoding to fail.  This can be worked around by mocking the transcoding function, but it gets convoluted and ugly.

Reply:
I'm not sure what to do here.  No code is un-covered.  I would argue that testing this logic in the context of the api method itself doesn't necessarily add anything.

Reply:
Okay, I've added logic to test the actual api method with default-valued required fields.